### PR TITLE
Add OLMo model

### PR DESCRIPTION
**Pull request by janimo**
Support for OLMo models

Note: Only the new ones with -hf suffix are supported, not the old ones built before OLMo was integrated into transformers, those are deprecated and have an extra package requirement.

https://huggingface.co/collections/allenai/olmo-suite-65aeaae8fe5b6b2122b46778

### CODE REVIEW

**Comment by merrymercy** [comment]
Thanks for the contribution. Can you add it here https://github.com/sgl-project/sglang/blob/f1088e0fc87027a18347c98c0319e54eabbe6a03/test/srt/models/test_generation_models.py#L52-L56
and run the test https://github.com/sgl-project/sglang/blob/f1088e0fc87027a18347c98c0319e54eabbe6a03/docs/en/model_support.md?plain=1#L16-L23?

**Comment by janimo** [comment]
I had to bump the error tolerance for decode for the test to pass. The resulting tokens are all the same.

File: python/sglang/srt/models/olmo.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+# Adapted from
+# https://github.com/vllm-project/vllm/blob/c7f2cf2b7f67bce5842fedfdba508440fe257375/vllm/model_executor/models/olmo.py#L1
+"""Inference-only OLMo model compatible with HuggingFace weights."""
+from typing import Iterable, List, Optional, Tuple
+
+import torch
+from torch import nn
+from transformers import OlmoConfig
+from vllm.distributed import get_tensor_model_parallel_world_size
+from vllm.model_executor.layers.linear import (
```
Comment (zhyncs) [inline python/sglang/srt/models/olmo.py:25]: @janimo Could you submit another PR to replace this with SGLang's linear?

Comment (janimo) [reply python/sglang/srt/models/olmo.py:25]: @zhyncs done #1696
